setup.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright 2016-2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from setuptools import Command, setup
+from setuptools import find_packages
+from unittest import TestLoader, TextTestRunner
+
+from os import path
+
+PROJECT = 'oca-decorators'
+SHORT_DESC = (
+    'This is a library of decorators for Odoo developers.'
+)
+README_FILE = 'README.rst'
+
+CLASSIFIERS = [
+    'Development Status :: 4 - Beta',
+    'Environment :: Console',
+    'Intended Audience :: Developers',
+    'License :: OSI Approved :: GNU Lesser General Public License v3 or '
+    'later (LGPLv3+)',
+    'Programming Language :: Python',
+    'Programming Language :: Python :: 2',
+    'Programming Language :: Python :: 2.7',
+    'Programming Language :: Python :: 3',
+    'Programming Language :: Python :: 3.4',
+    'Programming Language :: Python :: 3.5',
+    'Programming Language :: Python :: 3.6',
+    'Topic :: Software Development :: Libraries :: Python Modules',
+]
+
+setup_vals = {
+    'name': PROJECT,
+    'author': 'Odoo Community Association (OCA)',
+    'author_email': 'tools@odoo-community.org',
+    'description': SHORT_DESC,
+    'url': 'https://oca.github.io/%s' % PROJECT,
+    'download_url': 'https://github.com/OCA/%s' % PROJECT,
+    'license': 'LGPL-3',
+    'classifiers': CLASSIFIERS,
+}
+
+
+if path.exists(README_FILE):
+    with open(README_FILE) as fh:
+        setup_vals['long_description'] = fh.read()
+
+
+install_requires = []
+if path.exists('requirements.txt'):
+    with open('requirements.txt') as fh:
+        install_requires = fh.read().splitlines()
+
+
+class FailTestException(Exception):
+    """ It provides a failing build """
+    pass
+
+
+class Tests(Command):
```

Comment:
What's the purpose of this class?

Reply:
It discovers tests to run, then runs them. I prefer to have the isolated class, because our internal CI can inject [`XmlTestRunner`](https://pypi.python.org/pypi/XmlTestRunner/) for parse by JUnit

Reply:
@lasley IMO we must keep setup.py the must simple as possible and only put into this file the information required to package and setup the code. Moreover, we can simplify the content of `.travis.yml` to only use standard python tools. I'll propose you a PR.

Reply:
@lasley I've simplified the code to run tests with standard tools. https://github.com/laslabs/oca-decorators/pull/1. You can see the result in https://github.com/OCA/oca-decorators/pull/3 .
IMO the code is cleaner, easier to read and maintain.